src/mscorlib/src/System/String.cs

```diff
@@ -957,9 +975,31 @@ public String[] Split(char[] separator, int count, StringSplitOptions options)
             return SplitInternal(separator, count, options);
         }
 
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private unsafe String[] SplitInternal(char separator, int count, StringSplitOptions options)
+        {
+            char* pSeparators = stackalloc char[1];
+            pSeparators[0] = separator;
+            return SplitInternal(pSeparators, /*separatorsLength*/ 1, count, options);
+        }
+
         [ComVisible(false)]
+        [System.Security.SecuritySafeCritical]  // auto-generated
         internal String[] SplitInternal(char[] separator, int count, StringSplitOptions options)
         {
+            unsafe
+            {
+                fixed (char* pSeparators = separator)
+                {
+                    int separatorsLength = separator == null ? 0 : separator.Length;
+                    return SplitInternal(pSeparators, separatorsLength, count, options);
+                }
+            }
+        }
+
+        [System.Security.SecuritySafeCritical]  // auto-generated
+        private unsafe String[] SplitInternal(char* separators, int separatorsLength, int count, StringSplitOptions options)
```

Comment:
This should be SecurityCritical, the safety of this methods depends on up stack code being correct (i.e. if you pass in a bad `seperatorsLength` you violate memory safety)/